### PR TITLE
Bug 1168035 - [Stingray][Dashboard] Focus/unfocus widgets after moving

### PR DESCRIPTION
**Pull request by rexboy7**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=88c03aa85748559d26786c009e16ff4518a23810

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=58983479f5ad547a6d142e016416d90a61a759bf

File: shared/js/smart-screen/key_navigation_adapter.js [hunk 1]
```diff
@@ -22,6 +28,7 @@
       switch (evt.type) {
         case 'keydown':
         case 'keyup':
+        case this._customevent:
```
Comment (huchengtw-moz) [inline shared/js/smart-screen/key_navigation_adapter.js]: If we just put customevent falling through. We may have all events without prefix, like keyup. Is this what you want? I would suggest to let this module be aware of `mozbrowserbeforekeydown` and `mozbrowserbeforekeyup`, and use one flag to turn this on or off. Currently, we have mozbrowser(before|after)key(down|up) events. We should have the ability to separate them out. All users of this module should decide which action they want to do while processing different events.

File: tv_apps/dashboard/js/dashboard.js [hunk 2]
```diff
@@ -58,6 +73,7 @@
       var direction = document.body.dataset.activeDirection;
       this.widgets[direction] && this.widgets[direction].toggleExpand(false);
       document.body.dataset.activeDirection = '';
+      document.activeElement.blur();
```
Comment (huchengtw-moz) [inline tv_apps/dashboard/js/dashboard.js]: We should keep focus at dashboard after we blur somebody. So, please add something like: `mainSection.focus()`.

We had met an issue at smart-home. While we only blur someone and keep focus at smart-home, other apps under smart-home may have opportunity to steal focus.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8807c8545b412c17449403bc93a698ba7f85239d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e82d5ad6dcd0ea73dfe7d7440f19d5fd8ca9e2a9

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9e7b310a58d4440b20e0793eb8f00fabbc9ec53a

**Pull request merged**